fetchcode/purl2url/__init__.py

```diff
@@ -0,0 +1,134 @@
+# fetchcode is a free software tool from nexB Inc. and others.
+# Visit https://github.com/nexB/fetchcode for support and download.
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# http://nexb.com and http://aboutcode.org
+#
+# This software is licensed under the Apache License version 2.0.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at:
+# http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+from urllib.parse import urlparse
+
+
+def index_in_list(index, list):
+    return index < len(list)
+
+
+def build_bitbucket(name, namespace, version, qualifiers, subpath):
+
+    if not (name and namespace):
+        return
+
+    url = "https://bitbucket.org/{namespace}/{name}".format(
+        namespace=namespace, name=name
+    )
+    if version:
+        url = "{url}/src/{version}".format(url=url, version=version)
+    if subpath:
+        url = "{url}/{subpath}".format(url=url, subpath=subpath)
+    return url
+
+
+def build_cargo(name, namespace, version, qualifiers, subpath):
```

Comment:
Why do we only use a subset of the arguments here to build a url?

Reply:
Can you please explain a little bit, what should be done here, it will be helpful for me :sweat_smile:

Reply:
This function accepts 5 different arguments, but you only use a subset of these to build the URL. You need to either:
1) use all the args in some way
2) remove extraneous args that aren't used

It is confusing otherwise

Reply:
Okay I am getting your point :), I am passing a kwarg from the purl2url function and unpacking it here, can you suggest me a better alternative, I was also finding some way for that, but not able to do so

Reply:
Just make the function accept kwargs. See: https://www.digitalocean.com/community/tutorials/how-to-use-args-and-kwargs-in-python-3

There are other resources out there as well.

Reply:
Gotcha, thanks